packages/conform-dom/formdata.ts

```diff
@@ -128,11 +128,37 @@ export function setValue(
 						: {}
 				: valueFn(pointer[key]);
 
-		pointer[key] = newValue;
+		if (canAddProperty(pointer, key)) {
+			pointer[key] = newValue;
+		}
 		pointer = pointer[key];
 	}
 }
 
+function canAddProperty(
+	obj: Object | Array<unknown>,
+	property: string | number,
+): boolean {
+	const propertyStr =
+		typeof property === 'number' ? property.toString() : property;
+	if (Object.prototype.hasOwnProperty.call(obj, propertyStr)) {
+		const descriptor = Object.getOwnPropertyDescriptor(obj, propertyStr);
+		if (descriptor && !descriptor.writable) {
+			return false;
+		}
+	}
+	if (!Object.isExtensible(obj)) {
+		return false;
+	}
+	if (Object.isSealed(obj) && !(propertyStr in obj)) {
+		return false;
+	}
+	if (Object.isFrozen(obj) && !(propertyStr in obj)) {
+		return false;
+	}
+	return true;
+}
+
```

Comment:
I wanna understand better the impact of this change. Can you explain a bit more what each condition is trying to handle?

Reply:
As i mentioned in #670 currently we can get TypeError if we will try to assign property on immutable object. 
Since we validate and get data from those parameters that can be passed by users, including frauders. We wouldn't want to get errors in runtime because of a user who enters intentionally incorrect data. Of course, I can handle the error, but I would like not to get errors in the application's runtime.

Reply:
The same issue we can reproduce with formdata as well.